### DISCUSSION TITLE
When do we use Prisma Data Proxy?

### DISCUSSION THREAD

**Discussion by un-focused** [prisma/prisma#15511] [Q&A] [answered]
Hi,

I am considering launching my backend serverlessly. I saw the Prisma Data Proxy & was wondering what it is for? Reading the documentation, it looks like it is a proxy that prevents cold starts as it keeps connections open. Is that what its primary use is for?

Additionally, how would this be useful in a Server based architecture?

**Comment by austincrim** [comment, marked as answer]
Hey @un-focused 👋,

[From our docs](https://www.prisma.io/docs/data-platform/data-proxy),

> The benefits of using the Data Proxy include:
> - avoid exhausting the number of database connections or needlessly scaling up your database
> - avoid serverless cold starts by reusing existing database connections
> - reduced bundle size for frictionless serverless deployments

The primary use, in my opinion, is its connection pooling capabilities for serverless backends. Since serverless functions can scale basically infinitely, your relational database needs a strategy to handle scale beyond the traditional connection limits. The Data Proxy provides a plug-in solution that is database-agnostic.

It is less applicable in a "server-ful" architecture, but may still be useful depending on the server scaling strategy and anticipated traffic.

**Reply by un-focused** [reply]
Awesome! If I were to use something like CockroachDB serverless, would I require this proxy or could I depend on the pooling of CockroachDB?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
